integration/leader_test.go

```diff
@@ -0,0 +1,145 @@
+// +build integration
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/m3db/m3cluster/internal/etcdcluster"
+	"github.com/m3db/m3cluster/services"
+	"github.com/m3db/m3cluster/services/leader"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testIntegrationCluster struct {
+	*etcdcluster.Cluster
+
+	t *testing.T
+}
+
+func newTestIntegrationCluster(t *testing.T, opts *etcdcluster.Options) *testIntegrationCluster {
+	if opts == nil {
+		opts = &etcdcluster.Options{}
+	}
+
+	cluster, err := etcdcluster.New(*opts)
+	require.NoError(t, err)
+
+	tc := &testIntegrationCluster{Cluster: cluster, t: t}
+
+	tc.startAndWait(30 * time.Second)
+
+	return tc
+}
+
+func (tc *testIntegrationCluster) service(override string, ttl int) services.LeaderService {
+	opts := newTestOptions(override, ttl)
+	client, err := tc.Client()
+	require.NoError(tc.t, err)
+
+	svc, err := leader.NewService(client, opts)
+	require.NoError(tc.t, err)
+
+	return svc
+}
+
+func (tc *testIntegrationCluster) startAndWait(timeout time.Duration) {
+	err := tc.Start()
+	require.NoError(tc.t, err)
+
+	err = tc.WaitReady(timeout)
+	require.NoError(tc.t, err)
+}
+
+func newTestOptions(override string, ttl int) leader.Options {
+	sid := services.NewServiceID().
+		SetEnvironment("e1").
+		SetZone("z1").
+		SetName("s1")
+
+	return leader.NewOptions().
+		SetServiceID(sid).
+		SetOverrideValue(override).
+		SetTTL(ttl)
+}
+
+func TestIntegration_Simple(t *testing.T) {
```

Comment:
Wonder what happens if you create a client on the test cluster and shutdown the cluster, after long enough time (past ttl) can the client still keep the lock?